firmware/boron-ins-fsm/src/stateMachine.cpp

```diff
@@ -208,17 +212,19 @@ void state2_duration(){
     Log.warn("Door opened, session over, going to idle from state2_duration");
     publishStateTransition(2, 0, checkDoor.doorStatus, checkINS.iAverage);
     saveStateChange(2, 2);
+    hasDurationAlertBeenSent = 0; // session is over, so reset flag
     stateHandler = state0_idle;
-
   }
-  else if(millis() - state2_duration_timer >= state2_max_duration){
+  else if((millis() - state2_duration_timer >= state2_max_duration)
+          && !hasDurationAlertBeenSent){
```

Comment:
Was the line too long if this was all on one line?

Reply:
I put this on a new line out of habit, but you are right, it is short enough that it can go on the same line